blocks/library/button/index.js

```diff
@@ -16,70 +17,106 @@ import BlockControls from '../../block-controls';
 import ToggleControl from '../../inspector-controls/toggle-control';
 import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 import ColorPalette from '../../color-palette';
+import ContrastChecker from '../../contrast-checker';
 import InspectorControls from '../../inspector-controls';
 import BlockDescription from '../../block-description';
 
 const { attr, children } = source;
+const { getComputedStyle } = window;
 
-registerBlockType( 'core/button', {
-	title: __( 'Button' ),
+class ButtonBlock extends Component {
+	constructor() {
+		super( ...arguments );
 
-	icon: 'button',
+		this.containers = {};
+		this.fallbackColors = {};
 
-	category: 'layout',
+		this.state = {
+			fallbackBackgroundColor: undefined,
+			fallbackTextColor: undefined,
+		};
 
-	attributes: {
-		url: {
-			type: 'string',
-			source: attr( 'a', 'href' ),
-		},
-		title: {
-			type: 'string',
-			source: attr( 'a', 'title' ),
-		},
-		text: {
-			type: 'array',
-			source: children( 'a' ),
-		},
-		align: {
-			type: 'string',
-			default: 'none',
-		},
-		color: {
-			type: 'string',
-		},
-		textColor: {
-			type: 'string',
-		},
-	},
+		this.updateAlignment = this.updateAlignment.bind( this );
+		this.toggleClear = this.toggleClear.bind( this );
+		this.bindRef = this.bindRef.bind( this );
+	}
 
-	getEditWrapperProps( attributes ) {
-		const { align, clear } = attributes;
-		const props = {};
+	componentDidMount() {
+		this.grabColors();
+	}
 
-		if ( 'left' === align || 'right' === align || 'center' === align ) {
-			props[ 'data-align' ] = align;
+	componentDidUpdate() {
+		this.grabColors();
+	}
+
+	updateAlignment( nextAlign ) {
+		this.props.setAttributes( { align: nextAlign } );
+	}
+
+	toggleClear() {
+		const { attributes, setAttributes } = this.props;
+		setAttributes( { clear: ! attributes.clear } );
+	}
+
+	bindRef( node ) {
+		if ( ! node ) {
+			return;
 		}
 
-		if ( clear ) {
-			props[ 'data-clear' ] = 'true';
+		this.containers.background = node;
+		this.containers.text = node.querySelector( '[contenteditable="true"]' );
+	}
+
+	grabColors() {
+		const { background, text } = this.containers;
+		const { textColor, color } = this.props.attributes;
+		const { fallbackTextColorInState, fallbackBackgroundColorInState } = this.state;
```

Comment:
When are these ever set? :)

We can make do with just the truthiness of `this.state.fallback*ColorInState`.

Reply:
Yes my logic had a bug, but the results looked like the expected in tests so I did not notice, it was solved :)